--- mntzorro.v	2026-01-21 22:47:56
+++ /Users/midwan/Downloads/zz9000/scanlines/mntzorro.v	2026-04-08 23:55:04
@@ -199,6 +199,7 @@
    output reg [31:0] video_control_data_out,
    output reg [7:0]  video_control_op_out,
    output reg video_control_interlace_out,
+   output reg [7:0] scanline_intensity_out,
    input wire [1:0] video_control_vblank_in,
    
    // ZZ9000AX peripheral reset
@@ -973,6 +974,7 @@
   reg        video_control_vblank; // from input
   reg        video_control_hblank; // from input
   reg        video_control_interlace;
+  reg [7:0]  scanline_intensity = 8'h00;
 
   reg zorro_ram_read_flag;
   reg zorro_ram_write_flag ;
@@ -2259,6 +2261,7 @@
             'h02: video_control_data_zorro[15:0]  <= regdata_in[15:0];
             'h04: video_control_op_zorro[7:0]     <= regdata_in[7:0]; // FIXME
             'h06: videocap_mode_in <= regdata_in[0];
+            'h0a: scanline_intensity <= regdata_in[7:0];
             //'h08: E7M_RESET <= regdata_in[0];
             //'h0a: E7M_PWRDWN <= regdata_in[0];
             'h10: videocap_address[31:16] <= regdata_in[15:0];
@@ -2329,6 +2332,7 @@
     video_control_vblank   <= video_control_vblank_in[0];
     video_control_hblank   <= video_control_vblank_in[1];
     video_control_interlace_out <= video_control_interlace;
+    scanline_intensity_out <= scanline_intensity;
 
     // snoop the screen width for correct capture pitch
     if (video_control_op == 2) begin
